src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -47,12 +69,22 @@ internal override void CorrectCertificateState(X509Certificate2 candidate)
 
         protected override bool IsExportable(X509Certificate2 c) => true;
 
-        protected override void TrustCertificateCore(X509Certificate2 certificate) =>
-            throw new InvalidOperationException("Trusting the certificate is not supported on linux");
+        protected override void TrustCertificateCore(X509Certificate2 certificate)
+        {
+            foreach (var store in CertificateStores)
+            {
+                CertificateManager.Log.LinuxTrustCertificate(CertificateManager.GetDescription(certificate), store.StoreName);
+                store.TryInstallCertificate(certificate);
+                // TODO: handle failure.
+            }
+        }
 
         protected override void RemoveCertificateFromTrustedRoots(X509Certificate2 certificate)
         {
-            // No-op here as is benign
+            foreach (var store in CertificateStores)
+            {
+                store.DeleteCertificate(certificate);
+            }
```

Comment:
Each operation likely needs to go in a `try..catch` block and emit logs to indicate success or failure. If something goes wrong we want to know where we failed.

Reply:
Here is an example for reference:
https://github.com/dotnet/aspnetcore/blob/43241a3dc4b93f553226dd40ba3211891979d7e9/src/Shared/CertificateGeneration/MacOSCertificateManager.cs#L190-L227